Common/Product/SharedProject/ProjectNode.cs

```diff
@@ -2040,7 +2040,41 @@ internal enum EventTriggering {
             this.currentConfig = null;
         }
 
+        public virtual void SetOrAddPropertyAfter(string propertyName, string propertyValue, string afterProperty) {
+            Utilities.ArgumentNotNull("propertyName", propertyName);
+            Utilities.ArgumentNotNull("afterProperty", afterProperty);
+
+            Utilities.ArgumentNotNull("propertyName", propertyName);
+            Site.GetUIThread().MustBeCalledFromUIThread();
+
+            var oldValue = GetUnevaluatedProperty(propertyName) ?? string.Empty;
+            propertyValue = propertyValue ?? string.Empty;
+
+            if (oldValue.Equals(propertyValue, StringComparison.Ordinal)) {
+                // Property is unchanged or unspecified, so don't set it.
+                return;
+            }
 
+            // Check out the project file.
+            if (!this.QueryEditProjectFile(false)) {
+                throw Marshal.GetExceptionForHR(VSConstants.OLE_E_PROMPTSAVECANCELLED);
+            }
+
+            var oldProp = this.buildProject.GetProperty(propertyName);
+            if (oldProp == null || oldProp.IsImported) {
+                var propertyGroups = from g in this.buildProject.Xml.PropertyGroups
+                                     where g.Children.Where(c => c.ElementName == afterProperty).Any()
+                                     select g;
+                var group = propertyGroups.Any() ? propertyGroups.First() : this.buildProject.Xml.PropertyGroups.First();
```

Comment:
If it's defined in multiple property groups, should we add after the last group to get the correct value? Also, what if there are no property groups at all, `group` will be `null` and it will crash.
How about this?
```
var group = propertyGroups.LastOrDefault();
if (group != null) {
    var newProp = group.SetProperty(propertyName, propertyValue);
} else {
    this.buildProject.SetProperty(propertyName, propertyValue);
}
```

Reply:
I agree. I will update this.